File: BackEnd/FoodOrder.Core/ViewModels/Orders/OrderCreateVM.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodOrder.Core.ViewModels.Orders
{
    public class OrderCreateVM
    {
        [Required(ErrorMessage = "AppUserID is requied")]
        public Guid AppUserID { get; set; }
      
        public bool IsPaid { get; set; }
        public DateTime? DatePaid { get; set; }
        public int OrderStatusID { get; set; }
        public int? PromotionID { get; set; }
        public decimal? PromotionAmount { get; set; }

        public string AddressString { get; set; }
        public string AddressName { get; set; }

        //public List<int> CartIDs { get; set; }
    }
}

File: FrontEnd/food_order/lib/view_models/Orders/OrderCreateVM.dart
import 'package:json_annotation/json_annotation.dart';

part 'OrderCreateVM.g.dart';

@JsonSerializable(explicitToJson: true)
class OrderCreateVM {
  OrderCreateVM();
  OrderCreateVM.explicit({
    required this.appUserID,
    required this.isPaid,
    this.datePaid,
    required this.orderStatusID,
    this.promotionID,
    required this.promotionAmount,
    required this.addressString,
    required this.addressName,
  });
  late String appUserID;

  late bool isPaid;
  DateTime? datePaid;
  late int orderStatusID;
  int? promotionID;
  double? promotionAmount;

  late String addressString;
  late String addressName;

  //late List<int> cartIDs;

  factory OrderCreateVM.fromJson(Map<String, dynamic> json) =>
      _$OrderCreateVMFromJson(json);
  Map<String, dynamic> toJson() => _$OrderCreateVMToJson(this);
}
